src/Essentials/src/SecureStorage/SecureStorage.android.cs

```diff
@@ -1,395 +1,175 @@
-				{
-					System.Diagnostics.Debug.WriteLine($"Unable to unwrap key: Illegal Block Size. This may be caused by system backup or upgrades. All secure storage items will now be removed. {ibsEx.Message}");
-				}
-				catch (BadPaddingException paddingEx)
-				{
-					System.Diagnostics.Debug.WriteLine($"Unable to unwrap key: Bad Padding. This may be caused by system backup or upgrades. All secure storage items will now be removed. {paddingEx.Message}");
-				}
-				SecureStorage.RemoveAll();
 			}
-
-			var keyGenerator = KeyGenerator.GetInstance(aesAlgorithm);
-			var defSymmetricKey = keyGenerator.GenerateKey();
-
-			var newWrappedKey = WrapKey(defSymmetricKey, keyPair.Public);
-
-			Preferences.Set(prefsMasterKey, Convert.ToBase64String(newWrappedKey), alias);
-
-			return defSymmetricKey;
 		}
 
-		// API 23+ Only
-		ISecretKey GetSymmetricKey()
+		static ISharedPreferences GetEncryptedSharedPreferences()
 		{
-			Preferences.Set(useSymmetricPreferenceKey, true, SecureStorage.Alias);
+			var runtimeApiLevel = SdkInt;
+			
+			string fileName;
+			string prefsMainKey;
+			var context = Application.Context;
 
-			var existingKey = keyStore.GetKey(alias, null);
 
-			if (existingKey != null)
+			if (runtimeApiLevel >= 23)
 			{
-				var existingSecretKey = existingKey.JavaCast<ISecretKey>();
-				return existingSecretKey;
+				prefsMainKey = GetOrCreateMasterKeyApi23OrNewer();
+				fileName = "MAUI";
+			}
+			else
+			{
+				prefsMainKey = GetOrCreateMasterKeyApi22OrOlder(context);
+				fileName = "MAUI_Legacy";
 			}
 
-			var keyGenerator = KeyGenerator.GetInstance(KeyProperties.KeyAlgorithmAes, androidKeyStore);
-			var builder = new KeyGenParameterSpec.Builder(alias, KeyStorePurpose.Encrypt | KeyStorePurpose.Decrypt)
-				.SetBlockModes(KeyProperties.BlockModeGcm)
-				.SetEncryptionPaddings(KeyProperties.EncryptionPaddingNone)
-				.SetRandomizedEncryptionRequired(false);
-
-			keyGenerator.Init(builder.Build());
+			var sharedPreferences = EncryptedSharedPreferences.Create(
+				fileName,
+				prefsMainKey,
+				context,
+				EncryptedSharedPreferences.PrefKeyEncryptionScheme.Aes256Siv,
+				EncryptedSharedPreferences.PrefValueEncryptionScheme.Aes256Gcm
+			);
 
-			return keyGenerator.GenerateKey();
+			return sharedPreferences;
 		}
 
-		KeyPair GetAsymmetricKeyPair()
-		{
-			// set that we generated keys on pre-m device.
-			Preferences.Set(useSymmetricPreferenceKey, false, SecureStorage.Alias);
+		static string GetOrCreateMasterKeyApi23OrNewer()
+			=> MasterKeys.GetOrCreate(MasterKeys.Aes256GcmSpec);
 
-			var asymmetricAlias = $"{alias}.asymmetric";
+		static string GetOrCreateMasterKeyApi22OrOlder(Context context)
+		{
+			// older versions don't support symmetric keys. but we still want to always use the KeyStore
 
-			var privateKey = keyStore.GetKey(asymmetricAlias, null)?.JavaCast<IPrivateKey>();
-			var publicKey = keyStore.GetCertificate(asymmetricAlias)?.PublicKey;
+			var asymmetricAlias = $"dotnet.MAUI.asymmetric";
```

Comment:
Is there a reason to not use the shared `Alias` that contains the package name? Will there be conflicts at all if multiple apps use the essentials SS?

Reply:
Great questions! I'm not sure. Alias, I'd think not. As for conflicts across apps, that's an interesting question... which I'm also a bit confused by. Hoping to chat a bit more offline about that one